catalog/src/main/java/org/killbill/billing/catalog/DefaultVersionedCatalog.java

```diff
@@ -190,6 +190,21 @@ private void validatePlanShape(final Plan plan, final Plan targetPlan, final Val
                                                              cur.getName(), plan.getName(), plan.getCatalog().getEffectiveDate(), targetPlan.getCatalog().getEffectiveDate()),
                                                DefaultVersionedCatalog.class, ""));
             }
+	    //Fix for https://github.com/killbill/killbill/issues/1465
+            validatePlanDuration(plan, cur, errors);
+        }
+    }
+    
+    private void validatePlanDuration(final Plan plan, final PlanPhase cur, final ValidationErrors errors) {
+        if(cur.getPhaseType().name().equals(PhaseType.EVERGREEN.name()) && !cur.getDuration().getUnit().name().equals(TimeUnit.UNLIMITED.name())) {
+            errors.add(new ValidationError(String.format("EVERGREEN Phase '%s'for plan '%s' in version '%s'  must have duration as UNLIMITED'",
+                    cur.getName(), plan.getName(), plan.getCatalog().getEffectiveDate()),
+                    DefaultVersionedCatalog.class, "")); 
+        }
+        if(cur.getPhaseType().name().equals(PhaseType.FIXEDTERM.name()) && cur.getDuration().getUnit().name().equals(TimeUnit.UNLIMITED.name())) {
```

Comment:
I know this was not specified in the description of #1465 but it would probably make sense to also add this same validation for `PhaseType#TRIAL` and `PhaseType#DISCOUNT`. @pierre Any thoughts?

Reply:
Yup, that makes sense.

Reply:
@sbrossie Can you tell me the exact validations to be put for PhaseType#TRIAL and PhaseType#DISCOUNT .

Reply:
Validation for `TRIAL`  and `DISCOUNT` should be similar to `FIXEDTERM`

Reply:
This is done.